### PR TITLE
Adds Cryptographic Identity utility for working with various keys ect

### PR DESCRIPTION
**Pull request by danwt**
This is a utility set that I extracted from working on Key Assignment. 

It gives a really handy way to get all the various key and address types used by the sdk and tendermint, and their various interfaces, without having to piece everything together each time.

This is used extensively in the Key Assignment tests but I've opened this PR to get it in sooner, so that I can start using it to improve the diff tests ect.

### CODE REVIEW

**Review by smarshall-spitzbart — APPROVED** [review]
Great stuff! Could we delete [this struct ](https://github.com/cosmos/interchain-security/blob/main/testutil/keeper/unit_test_helpers.go#L180)and it's below helper in this PR? Should be easily replaceable with your more extensive helpers

File: testutil/crypto/crypto.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+package crypto
+
+import (
+	"encoding/binary"
+
+	"github.com/cosmos/ibc-go/v3/testing/mock"
```
[the remark this answers is not in the document]
Comment (danwt) [reply testutil/crypto/crypto.go:6]: Yeah good point! Will do Monday

**Comment by danwt** [comment]
> Great stuff! Could we delete [this struct ](https://github.com/cosmos/interchain-security/blob/main/testutil/keeper/unit_test_helpers.go#L180)and it's below helper in this PR? Should be easily replaceable with your more extensive helpers

@smarshall-spitzbart for sure! How about letting this one in first, then I (or we) can make a pass over the codebase and use these utils in all the obvious places?

**Comment by smarshall-spitzbart** [comment]
> > Great stuff! Could we delete [this struct ](https://github.com/cosmos/interchain-security/blob/main/testutil/keeper/unit_test_helpers.go#L180)and it's below helper in this PR? Should be easily replaceable with your more extensive helpers
> 
> @smarshall-spitzbart for sure! How about letting this one in first, then I (or we) can make a pass over the codebase and use these utils in all the obvious places?

Sure thing, a larger parse through the codebase would make sense when we have time 👍 , for now I've made https://github.com/cosmos/interchain-security/pull/473 which removes only the helper I was referencing. It also adds one more function, so maybe relevant to merge into this PR

**Review by MSalopek — APPROVED** [review]
LGTM!

This is really useful, needed it a couple of times already and had to do weird stuff :D

**Pull request merged**